pvlib/tests/iotools/test_psm3.py

```diff
@@ -21,7 +23,12 @@
     'Temperature Units', 'Pressure Units', 'Wind Direction Units',
     'Wind Speed', 'Surface Albedo Units', 'Version']
 PVLIB_EMAIL = 'pvlib-admin@googlegroups.com'
-DEMO_KEY = 'DEMO_KEY'
+try:
+    DEMO_KEY = os.environ["NREL_API_KEY"]
+except KeyError:
+    print("WARNING: NREL API KEY environment variable not set!")
+    print("Using DEMO_KEY instead. This may cause unexpected failures.")
```

Comment:
Please either using [`logging`](https://docs.python.org/3/library/logging.html) framework or [`warnings`](https://docs.python.org/3/library/warnings.html). _EG_:

```python
import warnings

# in try
warnings.warn("WARNING: NREL API KEY ...")
```

Or
```python
import logging

logging.basicConfig()
LOGGER = logging.grtLogger(__name__)
LOGGER.setLevel(logging.WARNING)

# in try
LOGGER.warning("WARNING: NREL API KEY ...")
```

I prefer warnings over logger, but I think either would work better than `print`, which is unpythonic because it blocks the main thread and is not customizable -- or: you can't silence it from the command line. EG you can silence warnings and logging from `python`

Reply:
I think `warnings.warn` is the better approach here. 

It would also be better if this lookup were part of a fixture that returns the api key rather than existing at the module level. That way you only get the warning if you're running a test that needs to access the live api.